### ISSUE TITLE
Exception causes are being removed.

### ISSUE THREAD

**Issue by loweryjk** [Kludex/starlette#1129]
I have code raising an exception from another exception.

```
try:
    return check_response(response)
except KeyError as exc:
    raise Exception('Failed to fetch the data.') from exc
```

I then have an exception handler.

```
@app.exception_handler(Exception)
async def exception(request: Request, exc: Exception):
    logger.info(exc)
    if exc.__cause__ is not None:
        logger.info(exc.__cause__)
    return self._json_response(HTTPStatus.INTERNAL_SERVER_ERROR, exc)
```
The second log statement never executes.

See this ticket for further details: https://github.com/tiangolo/fastapi/issues/2683

**Comment by oTree-org** [comment]
I think this is the same as this issue: https://github.com/encode/starlette/issues/1114

**Comment by JayH5** [comment]
Yeah this should be fixed by #1158
